### PR TITLE
Drop core-js, node 0.10 & 0.12 support; v0.5.0

### PR DESCRIPTION
**Pull request by gwicke**
/cc @wikimedia/services

### CODE REVIEW

File: lib/reqTemplate.js [hunk 1]
```diff
@@ -116,13 +115,12 @@ var globalMethods = {
         if (!isValidDate(date)) {
             var origDate = date;
 
-            if (typeof date === 'string' && /^\d+$/.test(date)) {
+            if (typeof date === 'string' && !Number.isNaN(date)) {
```
Comment (Pchelolo) [inline lib/reqTemplate.js:118]: Em, looks like you've based it on some older branch? This is totally unrelated

**Comment by d00rman** [comment]
Published as [v0.5.0](https://github.com/wikimedia/swagger-router/tree/v0.5.0)

**Comment by Pchelolo** [comment]
Ahhh, nooo, stupid GitHub reviews, I didn't push 'submit'. 

We need to revert this, since it somehow reverted my commit https://github.com/wikimedia/swagger-router/commit/fd87fc3ac63ed9bedb572fee4b517d19445fb514

**Comment by gwicke** [comment]
@Pchelolo: Sorry, it does indeed look like somehow I managed to un-do your change. Let me push a follow-up restoring that.

**Comment by gwicke** [comment]
Restore patch at https://github.com/wikimedia/swagger-router/pull/48
